lib/services/flowdock.ts

```diff
@@ -15,53 +15,239 @@ limitations under the License.
 */
 
 import * as Promise from 'bluebird';
+const Session = require('flowdock').Session;
 import * as _ from 'lodash';
 import * as path from 'path';
 import * as request from 'request-promise';
-import { FlowdockEmitRequestContext } from './flowdock-types';
-import { ServiceEmitContext, ServiceEmitRequest, ServiceEmitResponse, ServiceEmitter } from './service-types';
+import {
+    FlowdockMessage,
+    FlowdockMessageEmitContext,
+} from '../services/flowdock-types';
+import {
+    MessageEmitResponse,
+    MessageEvent,
+    MessageWorkerEvent,
+    ReceiptContext,
+} from '../utils/message-types';
+import {
+    MessageService
+} from './message-service';
+import {
+    ServiceEmitter,
+    ServiceListener,
+} from './service-types';
 
-// The Flowdock service currently allows a client to send data to an Inbox.
-export class FlowdockService implements ServiceEmitter {
-    private _serviceName = path.basename(__filename.split('.')[0]);
+/**
+ * Class for interacting with the Discourse API
+ * Is a MessageService, ServiceListener and ServiceEmitter
+ */
+export class FlowdockService extends MessageService implements ServiceEmitter, ServiceListener {
+    private static session = new Session(process.env.FD_SYSTEM_TOKEN);
```

Comment:
Full envvar names again please, `FLOWDOCK_SYSTEM_TOKEN`. Also, if these are actually API tokens, can they be called as such?